### PR TITLE
feat: add BitVec.[toInt_inj|toInt_ne]

### PR DESCRIPTION
**Pull request by tobiasgrosser**

### CODE REVIEW

File: src/Init/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -245,6 +245,12 @@ theorem eq_of_toInt_eq {i j : BitVec n} : i.toInt = j.toInt → i = j := by
   have _jlt := j.isLt
   split <;> split <;> omega
 
+theorem toInt_eq (x y : BitVec n) : x = y ↔ x.toInt = y.toInt :=
+  Iff.intro (congrArg BitVec.toInt) eq_of_toInt_eq
```
Comment (semorrison) [inline src/Init/Data/BitVec/Lemmas.lean:249]:
```suggestion
theorem toInt_inj (x y : BitVec n) : x.toInt = y.toInt ↔ x = y :=
```
per https://leanprover-community.github.io/contribute/naming.html#injectivity

File: src/Init/Data/BitVec/Lemmas.lean [hunk 2]
```diff
@@ -245,6 +245,12 @@ theorem eq_of_toInt_eq {i j : BitVec n} : i.toInt = j.toInt → i = j := by
...
+
+theorem toInt_ne (x y : BitVec n) : x ≠ y ↔ x.toInt ≠ y.toInt := by
+  rw [Ne, toInt_eq]
```
Comment (semorrison) [inline src/Init/Data/BitVec/Lemmas.lean:252]: Do you mind reversing this? We would usually write this as 
```
theorem toInt_ne_Int (x y : BitVec n) : x.toInt ≠ y.toInt ↔ x ≠ y := ...
```

**Comment by tobiasgrosser** [comment]
I addressed all comments. This now leads to an inconsistency with:

```lean
@[bv_toNat] theorem toNat_eq (x y : BitVec n) : x = y ↔ x.toNat = y.toNat :=
  Iff.intro (congrArg BitVec.toNat) eq_of_toNat_eq

@[bv_toNat] theorem toNat_ne (x y : BitVec n) : x ≠ y ↔ x.toNat ≠ y.toNat := by
  rw [Ne, toNat_eq]
```

would you like me to write a follow-up PR to address this inconsistency?

**Comment by semorrison** [comment]
> I addressed all comments. This now leads to an inconsistency with:
> 
> ```lean
> @[bv_toNat] theorem toNat_eq (x y : BitVec n) : x = y ↔ x.toNat = y.toNat :=
>   Iff.intro (congrArg BitVec.toNat) eq_of_toNat_eq
> 
> @[bv_toNat] theorem toNat_ne (x y : BitVec n) : x ≠ y ↔ x.toNat ≠ y.toNat := by
>   rw [Ne, toNat_eq]
> ```
> 
> would you like me to write a follow-up PR to address this inconsistency?

Oops, thanks for noticing that! Yes, such a PR would be great.

**Pull request merged**